client/src/createItemInputBox.tsx

```diff
@@ -92,34 +94,46 @@ export const CreateItemInputBox = ({
       loadingComponent={() => <span>Loading</span>}
       placeholder="enter message here..."
       value={message}
-      onChange={(event) => setMessage(event.target.value)}
-      onKeyPress={(event) =>
-        isEnterKey(event) && message && sendItem() && event.preventDefault()
-      }
+      onChange={(event) => {
+        event.target.style.height = "0";
+        // Chrome will sometimes show a scrollbar at the exact scrollHeight, so give it a .1px extra as a nudge not to add one
+        event.target.style.height = `${event.target.scrollHeight + 0.1}px`;
+        setMessage(event.target.value);
+      }}
+      onKeyPress={(event) => {
+        if (isEnterKey(event)) {
+          if (message) {
+            sendItem();
+          }
+          event.preventDefault();
+        }
+      }}
```

Comment:
although logically this hasn't changed (nice readability improvement btw), I spotted two problems when using in the grid...

1. pressing enter with a payload attached, but no message (which is valid and works when clicking the send button) - doesn't send - but I think it should - from the code, I assume this has always been a problem
2. grids own keyboard shortcuts seem to be firing whilst typing, e.g. `f` makes it go fullscreen - I suspect some use of `event.stopPropagation()` at the top of the event handler regardless of what key was pressed should probably fix that

Reply:
Good spots! These are fixed